### PR TITLE
Add SO_TYPE and SO_ACCEPTCONN handling in getsockopt

### PR DESCRIPTION
**Pull request by viraptor**

### CODE REVIEW

**Comment by viraptor** [comment]
Ping @carllerche (just in case you missed the PR, there's no comment for a few weeks)

**Comment by kamalmarhubi** [comment]
@carllerche ping for review. I haven't looked at any of the bits of nix that use macros. Of the current collaborators, you have the most recent commits in this area.

File: src/sys/socket/sockopt.rs [hunk 1]
```diff
@@ -317,4 +322,25 @@ mod test {
...
         assert!(a_cred.pid != 0);
     }
+
+    #[test]
+    fn is_socket_type_correct() {
+        use super::super::*;
+
+        let (a, _b) = socketpair(AddressFamily::Unix, SockType::Stream, 0, SockFlag::empty()).unwrap();
+        let a_type = getsockopt(a, super::SockType).unwrap();
+        assert!(a_type == SockType::Stream);
```
Comment (kamalmarhubi) [inline src/sys/socket/sockopt.rs]: can you add a `SockType::Datagram` or other test as well?

File: src/sys/socket/sockopt.rs [hunk 2]
```diff
@@ -317,4 +322,25 @@ mod test {
...
+    #[cfg(any(target_os = "freebsd",
+              target_os = "linux",
+              target_os = "nacl"))]
+    #[test]
+    fn can_get_listen_on_tcp_socket() {
+        use super::super::*;
+
+        let s = socket(AddressFamily::Inet, SockType::Stream, SockFlag::empty(), 0).unwrap();
+        let s_listening = getsockopt(s, super::AcceptConn).unwrap();
+        assert!(!s_listening);
```
Comment (kamalmarhubi) [inline src/sys/socket/sockopt.rs]: can you start listening and assert it returns true?

**Comment by kamalmarhubi** [comment]
Ok I finally decided to look at this and learn about `getsockopt` and figure out what the abstraction here is. Could you add a couple of extra test cases just to be sure it's doing the expected thing?

**Comment by viraptor** [comment]
Sure, I'll update soon.

**Comment by viraptor** [comment]
Added the requested tests

**Comment by kamalmarhubi** [comment]
That's great, thanks!

**Pull request merged**